### DISCUSSION TITLE
Voicebox - Why the coin flip?

### DISCUSSION THREAD

**Discussion by itsjamie** [lucidrains/voicebox-pytorch#20] [Q&A]
Hello! I've been following along on the implementation to try and learn machine learning while doing the fast.ai course and have been fascinated with this.

My question relates to the section of code that relates to this part of the paper.

> The audio/duration sequence is masked with pdrop = 0.3/0.2, and otherwise a segment of r% sequence length is masked, where
r ∼ U[70, 100]/U[10, 100].

I believe this is the related code:

```
# construct mask if not given
        if not exists(mask):
            if coin_flip():
                frac_lengths = torch.zeros((batch,), device = self.device).float().uniform_(*self.frac_lengths_mask)
                mask = mask_from_frac_lengths(seq_len, frac_lengths)
            else:
                mask = prob_mask_like((batch, seq_len), self.p_drop_prob, self.device)
```

One major aspect that I'm missing is why is there the 50% coin flip to choose between the segment masking, and the mask controlled by `pdrop`?

---

The other part that I wanted to make sure I understood is, what is "batch"?
Is "batch" representing each individual file and at inference time you would only have one "batch"?
Is it for handling training multiple files at one time, does it represent where they talk in the paper about training in batches?

---

Thanks for any help!

**Comment by itsjamie** [comment]
@lucidrains just don't want to pollute issues, but I'm not sure if you're looking here.
